src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -150,9 +150,19 @@ private Task InitializeWebHostRuntimeChannelsAsync()
             return Task.CompletedTask;
         }
 
-        private bool ShouldStartInPlaceholderMode()
+        internal bool ShouldStartInPlaceholderMode()
         {
-            return string.IsNullOrEmpty(_workerRuntime) && _environment.IsPlaceholderModeEnabled();
+            if (string.IsNullOrEmpty(_workerRuntime) && _environment.IsPlaceholderModeEnabled())
+            {
+                if (_environment.IsLinuxHostingEnvironment())
+                {
+                    return true;
+                }
+                // On Windows AppService Env, only start worker processes for legacy template site: FunctionsPlaceholderTemplateSite
+                string siteName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteName);
```

Comment:
recommend adding an environment extension for this, e.g. _environment.IsPlaceholderTemplateSite()

Reply:
thanks. Will update it.